pkg/compact/downsample/downsample_test.go

```diff
@@ -375,3 +374,94 @@ func (it *sampleIterator) Seek(int64) bool {
 func (it *sampleIterator) At() (t int64, v float64) {
 	return it.l[it.i].t, it.l[it.i].v
 }
+
+// memBlock is an in-memory block that implements a subset of the tsdb.BlockReader interface
+// to allow tsdb.instantWriter to persist the data as a block.
+type memBlock struct {
```

Comment:
so... it's not memBlock anymore, right? (: Can we name it properly?

Reply:
that's the same memBlock as before. I moved it to downsample_test, as it's used only there